packages/block-serialization-default-parser/parser.php

```diff
@@ -0,0 +1,272 @@
+        if ( 0 === $has_match ) {
+            return array( 'no-more-tokens' );
+        }
+
+        list( $match, $started_at ) = $matches[ 0 ];
+
+        $length     = strlen( $match );
+        $is_closer  = isset( $matches[ 'closer' ] ) && -1 !== $matches[ 'closer' ][ 1 ];
+        $is_void    = isset( $matches[ 'void' ] ) && -1 !== $matches[ 'void' ][ 1 ];
+        $namespace  = $matches[ 'namespace' ];
+        $namespace  = ( isset( $namespace ) && -1 !== $namespace[ 1 ] ) ? $namespace[ 0 ] : 'core/';
+        $name       = $namespace . $matches[ 'name' ][ 0 ];
+        $has_attrs  = isset( $matches[ 'attrs' ] ) && -1 !== $matches[ 'attrs' ][ 1 ];
+        $attrs      = $has_attrs ? json_decode( $matches[ 'attrs' ][ 0 ] ) : null;
+
+        /*
+         * This state isn't allowed
+         * This is an error
+         */
+        if ( $is_closer && ( $is_void || $has_attrs ) ) {
+            // we can ignore them since they don't hurt anything
+        }
+
+        if ( $is_void ) {
+            return array( 'void-block', $name, $attrs, $started_at, $length );
+        }
+
+        if ( $is_closer ) {
+            return array( 'block-closer', $name, null, $started_at, $length );
+        }
+
+        return array( 'block-opener', $name, $attrs, $started_at, $length );
+    }
+
+    function add_freeform( $length = null ) {
+        $length = $length ?: strlen( $this->document ) - $this->offset;
+
+        if ( 0 === $length ) {
+            return;
+        }
+
+        $this->output[] = array(
+            'attrs'     => new stdClass(),
+            'innerHTML' => substr( $this->document, $this->offset, $length ),
+        );
+    }
+
+    function add_inner_block(BSDP_Block $block, $token_start, $token_length, $last_offset = null ) {
+        $parent = $this->stack[ count( $this->stack ) - 1 ];
+        $parent->block->innerBlocks[] = $block;
+        $parent->block->innerHTML .= substr( $this->document, $parent->prev_offset, $token_start - $parent->prev_offset );
+        $parent->prev_offset = $last_offset ?: $token_start + $token_length;
+    }
+
+    function add_block_from_stack( $end_offset = null ) {
+        $stack_top = array_pop( $this->stack );
+        $prev_offset = $stack_top->prev_offset;
+
+        $stack_top->block->innerHTML .= isset( $end_offset )
+            ? substr( $this->document, $prev_offset, $end_offset - $prev_offset )
+            : substr( $this->document, $prev_offset );
+
+        if ( isset( $stack_top->leading_html_start ) ) {
+            $this->output[] = array(
+                'attrs' => array(),
+                'innerHTML' => substr(
+                    $this->document,
+                    $stack_top->leading_html_start,
+                    $stack_top->token_start - $stack_top->leading_html_start
+                ),
+            );
+        }
+
+        $this->output[] = $stack_top->block;
+    }
+}
+
+function bdsp_select_parser( $prev_parse_class ) {
+    return 'BSDP_Parser';
```

Comment:
There's a typo at _BSDP_. Anyway, given that the `apply_filters` call in `gutenberg_parse_blocks` defaults to `'BDSP_Parser'`, we should remove this bit.

Reply:
good catch! Is removed the function in 9c85a60cf